### PR TITLE
Event api key auth 683/684

### PR DESCRIPTION
**Pull request by bingzhang**
## Description
_Please provide a summary of the pull request and the issue it fixes. Please add necessary details, context, dependencies, explanation of when review is needed (see next section), etc._

**Fixes #(add issue number here and remove parentheses)**

[comment]: # (This project only accepts pull requests related to open issues.)
[comment]: # (If suggesting a new feature or change, please discuss it in an issue first.)
[comment]: # (If fixing a bug, there should be an issue describing it with steps to reproduce.)

## Review Time Estimate
_Please give your idea of how soon this pull request needs to be reviewed by selecting one of the options below. This can be based on the criticality of the issue at hand and/or other relevant factors._

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Immediately
- [ ] Within a week
- [x] When possible

## Type of changes
_Please select a relevant option:_

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Other (any another change that does not fall in one of the above categories.)

## Checklist:
_Please select all applicable options:_

[comment]: # (To select your options, please put an 'x' in the all boxes that apply.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] I have signed the Rokwire Contributor License Agreement ([CLA](https://rokwire.org/rokwire_cla)). (_Any contributor who is not an employee of the University of Illinois whose official duties include contributing to the Rokwire software, or who is not paid by the Rokwire project, needs to sign the CLA before their contribution can be accepted._)
- [ ] I have updated the [CHANGELOG](../CHANGELOG.md).
- [ ] I have read the [Contributor Guidelines](../CONTRIBUTING.md).
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My change requires updating the documentation.
- [ ] I have made necessary changes to the documentation.
- [ ] I have added tests related to my changes.
- [ ] My changes generate no new warnings.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

---

[comment]: # (Template credit: This pull request template is based on Embedded Artistry {https://github.com/embeddedartistry/templates/blob/master/.github/PULL_REQUEST_TEMPLATE.md}, Clowder {https://github.com/clowder-framework/clowder/blob/develop/.github/PULL_REQUEST_TEMPLATE.md}, and TalAter {https://github.com/TalAter/open-source-templates} templates.)

### CODE REVIEW

**Comment by bingzhang** [comment]
@sandeep-ps it passed the test cases in my local db. need more tests though.

File: eventservice/api/controllers/events.py [hunk 1]
```diff
@@ -41,10 +43,28 @@
...
+    if 'user_token_data' in g:
+        include_private_events = True
+        auth_resp = g.user_token_data
+        uin = auth_resp.get('uiucedu_uin')
+        url = "%s%s/groups" % (cfg.GROUPS_BUILDING_BLOCK_ENDPOINT, uin)
+        headers = {"Content-Type": "application/json",
+                  "ROKWIRE_GS_API_KEY": cfg.ROKWIRE_GROUPS_API_KEY}
+
+        req = requests.get(url, headers=headers)
+        if req.status_code == 200:
```
Comment (sandeep-ps) [inline eventservice/api/controllers/events.py:58]: If there is an error communicating with Groups Building Block, Events Building Block should return HTTP 500 with an appropriate message in the response body.

File: eventservice/api/controllers/events.py [hunk 2]
```diff
@@ -72,9 +92,15 @@ def _get_events_result(query, limit, skip):
...
-    for cond in query.get('$and'):
-        if cond.get('isSuperEvent'):
-            is_super_event = True
+    if query.get('or'):
+        for subquery in query:
+            for cond in subquery.get('$and'):
+                if cond.get('isSuperEvent'):
+                    is_super_event = True
+    else:
+        for cond in query.get('$and'):
```
Comment (sandeep-ps) [inline eventservice/api/controllers/events.py:101]: I am getting an internal server error here when doing GET /events?category=Entertainment
```
2021-05-27T15:21:11.253Z ERROR   [Thread-2  ] : events_building_block - 'NoneType' object is not iterable
Traceback (most recent call last):
  File "rokwire-building-blocks-api/eventservice/api/controllers/events.py", line 72, in search
    result, result_len = _get_events_result(
  File "rokwire-building-blocks-api/eventservice/venv/lib/python3.8/site-packages/diskcache/recipes.py", line 426, in wrapper
    pair = timer(*args, **kwargs)
  File "rokwire-building-blocks-api/eventservice/venv/lib/python3.8/site-packages/diskcache/recipes.py", line 385, in timer
    result = func(*args, **kwargs)
  File "rokwire-building-blocks-api/eventservice/api/controllers/events.py", line 101, in _get_events_result
    for cond in query.get('$and'):
TypeError: 'NoneType' object is not iterable

```

**Review by sandeep-ps — APPROVED** [review]

**Review by sandeep-ps — CHANGES_REQUESTED** [review]
@bingzhang, please update the CHANGELOG. Thanks.

File: CHANGELOG.md [hunk 3]
```diff
@@ -24,7 +24,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Add release script for Contributions Building Block.[#675](https://github.com/rokwire/rokwire-building-blocks-api/issues/675)
 - Add okay endpoint to Contributions Building Block.[#679](https://github.com/rokwire/rokwire-building-blocks-api/issues/679)
 - Add isGroupPrivate to Events schema.[#682](https://github.com/rokwire/rokwire-building-blocks-api/issues/682)
-
+- Add group authentication to events building block.[#691](https://github.com/rokwire/rokwire-building-blocks-api/issues/683, https://github.com/rokwire/rokwire-building-blocks-api/issues/684)
```
Comment (sandeep-ps) [inline CHANGELOG.md:27]:
```suggestion
- Add group authorization to events building block. [#683](https://github.com/rokwire/rokwire-building-blocks-api/issues/683), [#684](https://github.com/rokwire/rokwire-building-blocks-api/issues/684)
```

**Review by sandeep-ps — APPROVED** [review]

**Pull request merged**